src/app/qgisapp.cpp

```diff
@@ -4761,7 +4761,7 @@ QgsMapCanvasDockWidget *QgisApp::createNewMapCanvasDock( const QString &name, bo
   connect( mapCanvasWidget->dockableWidgetHelper(), &QgsDockableWidgetHelper::closed, this, [ this, mapCanvasWidget]
   {
     mOpen2DMapViews.remove( mapCanvasWidget );
-    delete mapCanvasWidget;
+    mapCanvasWidget->deleteLater();
```

Comment:
Is this relevant? Shouldn't that be a separate backport?

Reply:
Good catch! Other commits slipped in...   :/    I removed them.
 Thanks